src/link/edit.js

```diff
@@ -100,6 +123,16 @@ const Edit = ( {
 					onClose={ stopAddingPreview }
 				/>
 			) }
+			{ ( viewingPreview && !addingPreview ) && (
+				<PreviewEditUI
+					anchorRef={ anchorRef }
+					title={ previewTitle }
+					onClose={ stopViewingPreview }
```

Comment:
I can see some weird popup behaviour when choosing one then another

after trying with this code it looks fine, can you try if it works for you?
```suggestion
			{ ( viewingPreview && !addingPreview && isActive ) && (
				<PreviewEditUI
					anchorRef={ anchorRef }
					title={ previewTitle }
```

Reply:
> weird popup behaviour when choosing one then another

I think I can see it too, what I'm seeing is:

* selecting one (first) preview in one paragraph/block, and then clicking outside on screen to close that first preview, and then clicking on a second preview (regardless of paragraph/block)---> ✅ it transitions fine 
* selecting one (first) preview in one paragraph/block, and then (without clicking outside on screen to close first preview) clicking on a second preview in a different paragraph/block---> ✅ it transitions fine 
* selecting one (first) preview in one paragraph/block, and then (without clicking outside on screen to close first preview) clicking on a second preview in the _same_ paragraph/block---> 🐛 transition fails, I think I can see it opening and then immediately closing. 

Same for you? With your code suggestion though I think it's almost there but not quite yet

Reply:
Narrowed it down to the `onClose` handler for `PreviewEditUI` in my setup: https://github.com/wikimedia/wikipediapreview-wordpress/pull/30/commits/83ffd968d9925a4d66c2ef20580d2ec8944dfe19, let me know if that doesn't work for you

Reply:
Yes, it works for me now,

> selecting one (first) preview in one paragraph/block, and then (without clicking outside on screen to close first preview) clicking on a second preview in the same paragraph/block---> 🐛 transition fails, I think I can see it opening and then immediately closing.

After the changes, I can see there is a "blink" when the transition. Do you know why?

this "blink" doesn't existed in the second point. (clicking on different paragraph/block)

Reply:
Haven't been able to figure out why exactly, it seems there is difference in what happens under the hood of the wordpress/rich-text logic when you click within the same block or outside/different block. I'd say the blink is not a deal-breaker but certainly annoying 😑

Reply:
I know why there is "blink" when having new title, because of the event `onClosePreview` that `stopViewingPreview` then `startViewingPreview` immediately. looking into it now.